src/Symfony/Component/DependencyInjection/EnvVarProcessor.php

```diff
@@ -341,6 +343,22 @@ public function getEnv(string $prefix, string $name, \Closure $getEnv): mixed
             return trim($env);
         }
 
+        if ('date_time' === $prefix) {
+            try {
+                return new \DateTime($env);
+            } catch (\Exception) {
+                throw new RuntimeException(sprintf('Env var "%s" is not a valid date time string.', $name));
+            }
+        }
+
+        if ('date_time_immutable' === $prefix) {
+            try {
+                return new \DateTimeImmutable($env);
```

Comment:
You should use the [Clock component](https://symfony.com/doc/current/components/clock.html) when available to support mockable `now`.

Reply:
I'm not sure I understand you correctly. Clock component (as I see in documentation) can return only ->now(), so I can change the code to use Clock::now() (if available) when the $env === 'now' but in other cases I still need to use \DateTimeImmutable, right?

Reply:
The clock allows to set the default timezone.
You can find some conversion logic in `DateTimeValueResolver`.

https://github.com/symfony/symfony/blob/2bac801f8a4cba02e8ecc840fa7bf512034f860a/src/Symfony/Component/HttpKernel/Controller/ArgumentResolver/DateTimeValueResolver.php#L86

Reply:
Done. I hope I did what you expected.